src/datapilot/core/platforms/dbt/insights/checks/check_macro_args_have_desc.py

```diff
@@ -0,0 +1,77 @@
+from typing import List
+from typing import Tuple
+
+from datapilot.core.insights.utils import get_severity
+from datapilot.core.platforms.dbt.insights.checks.base import ChecksInsight
+from datapilot.core.platforms.dbt.insights.schema import DBTInsightResult
+from datapilot.core.platforms.dbt.insights.schema import DBTModelInsightResponse
+from datapilot.core.platforms.dbt.schemas.manifest import AltimateResourceType
+
+
+class CheckMacroArgsHaveDesc(ChecksInsight):
+    NAME = "Check Macro Args Have Desc"
+    ALIAS = "check_macro_args_have_desc"
+    DESCRIPTION = (
+        "Checks that macros have descriptions for their arguments. Descriptions help in understanding the purpose of the macro arguments."
+    )
+    REASON_TO_FLAG = "Macros without descriptions for their arguments can lead to confusion and hinder effective data modeling and analysis. It's important to have descriptions for macro arguments."
+
+    def _build_failure_result(
+        self,
+        node_id: str,
+    ) -> DBTInsightResult:
+        """
+        Build failure result for the insight if a macro doesn't have a description.
+
+        :return: An instance of InsightResult containing failure message and recommendation.
+        """
+
+        failure_message = f"The macro `{node_id}` does not have a description."
+        recommendation = "Add a description to the macro to help in understanding the purpose of the macro."
+
+        return DBTInsightResult(
+            type=self.TYPE,
+            name=self.NAME,
+            message=failure_message,
+            recommendation=recommendation,
+            reason_to_flag=self.REASON_TO_FLAG,
+        )
+
+    def generate(self, *args, **kwargs) -> List[DBTModelInsightResponse]:
+        """
+        Generate a list of InsightResponse objects for each model in the DBT project,
+        identifying macros whose arguments don't have descriptions.
+        :return: A list of InsightResponse objects.
+        """
+
+        insights = []
+        for node_id, node in self.nodes.items():
```

Comment:
I don't think this works, We will need to create self.macros()/ We need to parse this from ManifestV11.

https://github.com/yu-iskw/dbt-artifacts-parser/blob/main/dbt_artifacts_parser/parsers/manifest/manifest_v11.py#L1442